kratos/utilities/intersection_utilities.h

```diff
@@ -232,6 +232,56 @@ class IntersectionUtilities
         return 0;
     }
 
+    /**
+     * Find the 3D intersection of a plane (no boundaries) with an edge (bounded line)
+     * @param rPlaneBasePoint Base point of the plane to intersect
+     * @param rPlaneNormal Normal vector of the plane to intersect
+     * @param rEdgePoint1 Coordinates of the first point of the intersecting edge
+     * @param rEdgePoint2 Coordinates of the second point of the intersecting edge
+     * @param rIntersectionPoint The intersection point coordinates
+     * @return The intersection type index:
+     * 0 (parallel or out of bounds - no intersection)
+     * 1 (unique intersection point)
+     * 2 (edge and plane coincide - no intersection)
+     */
+    static int ComputePlaneEdgeIntersection(
+        const array_1d<double,3>& rPlaneBasePoint,
+        const array_1d<double,3>& rPlaneNormal,
+        const array_1d<double,3>& rEdgePoint1,
+        const array_1d<double,3>& rEdgePoint2,
+        array_1d<double,3>& rIntersectionPoint,
+        const double epsilon = 1e-12)
```

Comment:
Why do we need this function? Couldn't we use the already existent plane-line intersection one?

Reply:
Maybe, we could call `ComputePlaneEdgeIntersection` in the already previously existing `ComputeTriangleLineIntersection` intersection (I guess that's the one you mean and I didn't want to change an existing, working function right away.)
Because `ComputeTriangleLineIntersection` does the same as  `ComputePlaneEdgeIntersection` and then additionally checks whether the intersection point lies inside the triangle, as a triangle is given as input, not a plane.

Reply:
mmm... I think I got your point. Then I'd generalize the name to `ComputePlaneSegmentIntersection`.

Just to be completely sure about which the difference is, in here, if there is intersection, the point always lies in the plane, as we consider it infinite (this is not the case in the triangle case). Also we consider in here a segment, which has beginning and end (in the method I was referring to we consider an infinite line). Am I wrong?

BTW, if this is the case, we should clearly explain this in the function documentation.

Reply:
The first part is correct, infinite plane in contrast to inside a triangle. Both methods consider a segment and not an endless line, that's why I called it Edge. 
I could rename them to `ComputePlaneSegmentIntersection` and `ComputeTriangleSegmentIntersection`. There is also ComputeLineLineIntersection, which also checks the limits of the lines, or rather segments.

Reply:
OK. Then I'd name the new one according to the existent ones, that is `ComputePlaneLineIntersection`. Even though the other names are a bit misleading I'd leave them as they are right now (this would need to be discussed in a standalone PR).